Exiled.Events/Patches/Events/Player/InteractingDoor.cs

```diff
@@ -34,91 +45,127 @@ private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstructi
+
+                        // InteractingDoor.Process(__instance, ply, colliderId, ev);
+                        new(OpCodes.Ldarg_0),
+                        new(OpCodes.Ldarg_1),
+                        new(OpCodes.Ldarg_2),
+                        new(OpCodes.Ldloc_S, ev.LocalIndex),
+                        new(OpCodes.Call, Method(typeof(InteractingDoor), nameof(InteractingDoor.Process))),
+                });
 
-                    // Handlers.Player.OnInteractingDoor(ev);
-                    new(OpCodes.Ldloc_S, ev.LocalIndex),
-                    new(OpCodes.Dup),
-                    new(OpCodes.Call, Method(typeof(Handlers.Player), nameof(Handlers.Player.OnInteractingDoor))),
+            offset = -6;
+            index = newInstructions.FindIndex(instruction => instruction.Calls(PropertySetter(typeof(DoorVariant), nameof(DoorVariant.NetworkTargetState)))) + offset;
 
-                    // if (!ev.IsAllowed)
-                    //    go to permission denied;
-                    new(OpCodes.Callvirt, PropertyGetter(typeof(InteractingDoorEventArgs), nameof(InteractingDoorEventArgs.IsAllowed))),
-                    new(OpCodes.Brfalse_S, permissionDenied),
+            newInstructions.RemoveRange(index, 19);
+            newInstructions.InsertRange(
+                index,
+                new CodeInstruction[]
+                {
+                        // if (flag)
+                        // goto skip
+                        new(OpCodes.Brtrue_S, skip),
+
+                        // ev.InteractionResult = DoorBeepType.PermissionDenied;
+                        new(OpCodes.Ldloc_S, ev.LocalIndex),
+                        new(OpCodes.Ldc_I4_0),
+                        new(OpCodes.Callvirt, PropertySetter(typeof(InteractingDoorEventArgs), nameof(InteractingDoorEventArgs.InteractionResult))),
+
+                        // skip
+                        // InteractingDoor.Process(__instance, ply, colliderId, ev);
+                        new CodeInstruction(OpCodes.Ldarg_0).WithLabels(skip),
+                        new(OpCodes.Ldarg_1),
+                        new(OpCodes.Ldarg_2),
+                        new(OpCodes.Ldloc_S, ev.LocalIndex),
+                        new(OpCodes.Call, Method(typeof(InteractingDoor), nameof(InteractingDoor.Process))),
                 });
 
-            newInstructions[newInstructions.Count - 1].labels.Add(retLabel);
-
             for (int z = 0; z < newInstructions.Count; z++)
                 yield return newInstructions[z];
 
             ListPool<CodeInstruction>.Pool.Return(newInstructions);
         }
 
-        private static bool CanStateChange(DoorVariant variant)
+        private static void Process(DoorVariant __instance, ReferenceHub ply, byte colliderId, InteractingDoorEventArgs ev)
         {
-            return !(variant.GetExactState() > 0f && variant.GetExactState() < 1f);
+            Handlers.Player.OnInteractingDoor(ev);
+            if (ev.IsAllowed)
```

Comment:
i am really not sure about this because mostly every plugin use this to just DeniedPermission

Reply:
and even in a breaking change update it's will be confusing for people